src/hooks/useActionCenterSubscriber/useLimitOrderActionSubscriber.tsx

```diff
@@ -0,0 +1,394 @@
+  const buyAsset = useAppSelector(selectActiveQuoteBuyAsset)
+  const sellAmountCryptoBaseUnit = useAppSelector(selectInputSellAmountCryptoBaseUnit)
+  const buyAmountCryptoBaseUnit = useAppSelector(selectBuyAmountCryptoBaseUnit)
+  const quoteId = useAppSelector(selectActiveQuoteId)
+  const { currentData: ordersResponse } = useLimitOrdersQuery()
+  const toast = useToast({
+    render: ({ title, status, description, onClose }) => {
+      const handleClick = () => {
+        onClose()
+        onDrawerOpen()
+      }
+
+      const toastSx = {
+        '&:hover': {
+          cursor: 'pointer',
+          background: status === 'success' ? 'green.300' : 'red.300',
+        },
+      }
+
+      return (
+        <Flex
+          // We can't memo this because onClose prop comes from the render props
+          // eslint-disable-next-line react-memo/require-usememo
+          onClick={handleClick}
+          background={status === 'success' ? 'green.500' : 'red.500'}
+          color='white'
+          px={4}
+          py={2}
+          borderRadius='md'
+          // eslint-disable-next-line react-memo/require-usememo
+          sx={toastSx}
+        >
+          <Box py={1} me={2}>
+            {status === 'success' ? (
+              <Icon color={toastColor} as={TbCircleCheckFilled} height='20px' width='20px' />
+            ) : (
+              <Icon color={toastColor} as={TbCircleXFilled} height='20px' width='20px' />
+            )}
+          </Box>
+          <Box>
+            <CText fontWeight='bold' color={toastColor}>
+              {title}
+            </CText>
+            <CText color={toastColor}>{description}</CText>
+          </Box>
+        </Flex>
+      )
+    },
+  })
+  const openLimitOrders = useAppSelector(selectOpenLimitOrderActionsFilteredByWallet)
+  const quoteExpirationTimestamp = useAppSelector(selectActiveQuoteExpirationTimestamp)
+
+  const orderSubmissionMetadataFilter = useMemo(() => {
+    return { cowSwapQuoteId: quoteId ?? 0 }
+  }, [quoteId])
+
+  const limitOrderSubmissionMetadata = useSelectorWithArgs(
+    selectLimitOrderSubmissionMetadata,
+    orderSubmissionMetadataFilter,
+  )
+
+  const previousLimitOrderState = usePrevious(limitOrderSubmissionMetadata?.state)
+  const previousLimitOrderId = usePrevious(limitOrderSubmissionMetadata?.limitOrder.orderId)
+  const limitPrice = useAppSelector(selectActiveQuoteLimitPrice)
+
+  // Create action after user confirmed the intent of placing a limit order
+  useEffect(() => {
+    if (!limitOrderSubmissionMetadata) return
+    if (!sellAsset) return
+    if (!buyAsset) return
+    if (!quoteId) return
+    if (!activeQuote) return
+
+    const accountId = activeQuote.params.accountId
+
+    if (
+      limitOrderSubmissionMetadata.state ===
+        LimitOrderSubmissionState.AwaitingLimitOrderSubmission &&
+      previousLimitOrderState !== LimitOrderSubmissionState.AwaitingLimitOrderSubmission
+    ) {
```

Comment:
q: is this enough or do we want to ensure that the current ID does not already exist before upserting? If we ever upsert twice with diff uuids, we could end up with problems